Makefile

```diff
@@ -1,21 +1,32 @@
-.PHONY: build-rust-optimizer build-workspace-optimizer build publish-rust-optimizer publish-workspace-optimizer publish
+.PHONY: build-rust-optimizer build-workspace-optimizer build create-rust-optimizer-builder use-rust-optimizer-builder publish-rust-optimizer publish-workspace-optimizer publish
 
 DOCKER_NAME_RUST_OPTIMIZER := "cosmwasm/rust-optimizer"
 DOCKER_NAME_WORKSPACE_OPTIMIZER := "cosmwasm/workspace-optimizer"
-DOCKER_TAG := 0.12.1
+DOCKER_TAG := 0.12.2
+
+# Build images locally for the host CPU architecture
 
 build-rust-optimizer:
 	docker build -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer .
 
 build-workspace-optimizer:
 	docker build -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer .
 
-publish-rust-optimizer: build-rust-optimizer
-	docker push $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG)
+build: build-rust-optimizer build-workspace-optimizer
 
-publish-workspace-optimizer: build-workspace-optimizer
-	docker push $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG)
+# Build multi-CPU architecture images and publish them. rust alpine images support the linux/amd64 and linux/arm64/v8 architectures.
 
-build: build-rust-optimizer build-workspace-optimizer
+create-rust-optimizer-builder:
+	docker buildx create --name rust-optimizer-builder
+
+# create-rust-optimizer-builder must be run before running use-rust-optimizer-builder for the first time
+use-rust-optimizer-builder:
+	docker buildx use rust-optimizer-builder
+
+publish-rust-optimizer: use-rust-optimizer-builder
+	docker buildx build --platform linux/amd64,linux/arm64/v8 -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer --push .
```

Comment:
I'm hitting
```
#41 [linux/arm64 builder  7/19] RUN wasm-opt --version
#41 0.070 .buildkit_qemu_emulator: /usr/local/bin/wasm-opt: Invalid ELF image for this architecture
#41 ERROR: process "/dev/.buildkit_qemu_emulator /bin/sh -c wasm-opt --version" did not complete successfully: exit code: 255

#34 [linux/arm64 base-optimizer 2/4] RUN apk update &&   apk add --no-cache musl-dev
#34 1.716 v3.14.2-5-gd4163d4c6c [https://dl-cdn.alpinelinux.org/alpine/v3.14/main]
#34 1.716 v3.14.2-8-gc07e220a82 [https://dl-cdn.alpinelinux.org/alpine/v3.14/community]
#34 1.716 OK: 14810 distinct packages available
#34 1.796 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/aarch64/APKINDEX.tar.gz
#34 2.287 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/aarch64/APKINDEX.tar.gz
#34 2.981 (1/1) Installing musl-dev (1.2.2-r3)
#34 3.748 OK: 116 MiB in 27 packages
#34 CANCELED
------
 > [linux/arm64 builder  7/19] RUN wasm-opt --version:
#41 0.070 .buildkit_qemu_emulator: /usr/local/bin/wasm-opt: Invalid ELF image for this architecture
```

for the ARM build. Which makes sense, as the arm64 build would require a proper `wasm-opt` binary.
Taking a look at https://github.com/WebAssembly/binaryen/releases/tag/version_101, it's not there.

So, that means finding a binary somewhere, or  building `wasm-opt` from source.

Reply:
Am I missing something here? I see in your description above that you successfully built both images.

Reply:
OK, I now know what's happening. Even when you are natively running the `linux/arm64` image, `wasm-opt` is being run in an emulated way (through qemu + binfmt_misc)!

Now that I've installed qemu + binfmt_misc support, I'm able to not only build, but also run an arm64 build on my amd64 machine.

So, basically, I'm running an arm64 image on a amd64 machine, and the binaries that are amd64 that are inside that image, are being supported transparently. Isn't this technology amazing?

That said, I think installing a arm64 binary would be better. I'm now (cross-) compiling it inside of my shiny new arm64 image, but, it takes some time to build.

**Update**: Just found out that `apk add binaryen` installs `wasm-opt` version 98(!).

Reply:
OK, I'll branch off your MR with some suggestions for re-organizing the `Makefile`, and loading native `wasm-opt` directly from `apk`.

Reply:
Is this a Mac ARM vs. Linux ARM difference? Mac might be able to run the x86 binary through the emulator.

Reply:
See #56. @harryscholes couldn't add you as a reviewer, but please take a look.

Reply:
> Is this a Mac ARM vs. Linux ARM difference? Mac might be able to run the x86 binary through the emulator.

Yes, exactly. When I tried this build initially, I didn't have qemu + binfmt_misc installed. That's why it failed.

Reply:
```bash
$ docker buildx inspect
Name:   rust-optimizer-multi
Driver: docker-container

Nodes:
Name:      rust-optimizer-multi0
Endpoint:  unix:///var/run/docker.sock
Status:    running
Platforms: linux/amd64, linux/386, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
$ 
```
:-D

Reply:
See https://medium.com/@artur.klauser/building-multi-architecture-docker-images-with-buildx-27d80f7e2408 in case you are curious.